### PR TITLE
Rework Miniforge inputs

### PR DESCRIPTION
**Pull request by bollwyvl**
- follow-on to #133
- addresses concerns from https://github.com/conda-incubator/setup-miniconda/pull/137#issuecomment-752682346
- [x] stop doing fragile github actions stuff
- [x] make `miniforge-version` the indicator for using `download-miniforge`
- [x] make `Miniforge3` the default `miniforge-variant`
- [x] update docs
- [x] update example 10 test

### CODE REVIEW

**Comment by goanpeca** [comment]
Is this ready for review @bollwyvl ?

File: action.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ inputs:
     description:
       "If provided, this variant of Miniforge will be downloaded and installed.
       Currently-known values:
-      - Miniforge3
+      - Miniforge3 (default)
```
Comment (goanpeca) [inline action.yml:23]: @bollwyvl we should add the default here no?

```yaml
    required: false
    default: "Miniforge3"	    
```

**Review by goanpeca — CHANGES_REQUESTED** [review]
I think the action yaml needs a small fix.

Otherwise LGTM :)

File: action.yml (hunk 1 above)
Comment (bollwyvl) [reply action.yml:23]: i'll take a look...

Comment (bollwyvl) [reply action.yml:23]: 510ef46  and 6689c3c

**Review by goanpeca — APPROVED** [review]
Awesome! Thanks a lot @bollwyvl 🥳 !

File: action.yml (hunk 1 above)
Comment (bollwyvl) [reply action.yml:23]: 6689c3c points out that we still have some incidental coupling between the various inputs... it still seems manageable, but i wish i could think of a way to _simply_ express these constraints more declaratively... adding something like JSON schema could do, but more deps...

**Pull request merged**